tests/components/traccar_server/test_config_flow.py

```diff
@@ -0,0 +1,189 @@
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    side_effect: Exception,
+    error: str,
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.traccar_server.config_flow.ApiClient.get_server",
+        side_effect=side_effect,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": error}
+
+    with patch(
+        "homeassistant.components.traccar_server.config_flow.ApiClient.get_server",
+        return_value={"id": "1234"},
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "1.1.1.1:8082"
+    assert result["data"] == {
+        CONF_HOST: "1.1.1.1",
+        CONF_PORT: "8082",
+        CONF_USERNAME: "test-username",
+        CONF_PASSWORD: "test-password",
+        CONF_SSL: False,
+        CONF_VERIFY_SSL: True,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_options(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test options flow."""
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={},
+    )
+
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+
+    assert CONF_MAX_ACCURACY not in config_entry.options
+
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={CONF_MAX_ACCURACY: 2.0},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert config_entry.options == {
+        CONF_MAX_ACCURACY: 2.0,
+        CONF_EVENTS: [],
+        CONF_CUSTOM_ATTRIBUTES: [],
+        CONF_SKIP_ACCURACY_FILTER_FOR: [],
+    }
+
+
+async def test_abort_already_configured(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test abort for existing server."""
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: "1.1.1.1", CONF_PORT: "8082"},
+    )
+
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+
```

Comment:
I don't think you have to necessarily setup the entry to check for already_configured.

Reply:
Probably not, but it does not hurt.